modules/jdbc/src/test/java/org/apache/ignite/internal/jdbc/JdbcResultSetBaseSelfTest.java

```diff
@@ -0,0 +1,969 @@
+
+            expectNotSupported(rs::deleteRow);
+
+            expectNotSupported(rs::insertRow);
+
+            expectNotSupported(rs::moveToInsertRow);
+        }
+    }
+
+    @Test
+    public void navigationMethods() throws SQLException {
+        try (ResultSet rs = createResultSet(null,
+                List.of(new ColumnDefinition("C", ColumnType.STRING, 3, 0, false)),
+                List.of(List.of("A"), List.of("B")))
+        ) {
+            assertTrue(rs.isBeforeFirst());
+            assertFalse(rs.isAfterLast());
+            assertFalse(rs.isFirst());
+            assertFalse(rs.isLast());
+            assertEquals(0, rs.getRow());
+
+            assertTrue(rs.next());
+
+            assertFalse(rs.isBeforeFirst());
+            assertFalse(rs.isAfterLast());
+            assertTrue(rs.isFirst());
+            assertFalse(rs.isLast());
+            assertEquals(1, rs.getRow());
+
+            assertTrue(rs.next());
+
+            assertFalse(rs.isBeforeFirst());
+            assertFalse(rs.isAfterLast());
+            assertFalse(rs.isFirst());
+            assertTrue(rs.isLast());
+            assertEquals(2, rs.getRow());
+
+            assertFalse(rs.next());
+
+            assertFalse(rs.isBeforeFirst());
+            assertTrue(rs.isAfterLast());
+            assertFalse(rs.isFirst());
+            assertFalse(rs.isLast());
+            assertEquals(0, rs.getRow());
+        }
+
+        try (ResultSet rs = createResultSet(null,
+                List.of(new ColumnDefinition("C", ColumnType.STRING, 3, 0, false)),
+                List.of(List.of("A"), List.of("B")))
+        ) {
+            String error = "Result set is forward-only.";
+
+            expectSqlException(rs::first, error);
+            expectSqlException(rs::afterLast, error);
+            expectSqlException(rs::last, error);
+
+            assertTrue(rs.next());
+
+            expectSqlException(rs::first, error);
+            expectSqlException(rs::afterLast, error);
+            expectSqlException(rs::last, error);
+
+            expectSqlException(() -> rs.absolute(0), error);
+            expectSqlException(() -> rs.relative(0), error);
+            expectSqlException(rs::previous, error);
+
+            assertEquals(ResultSet.FETCH_FORWARD, rs.getFetchDirection());
+            rs.setFetchDirection(ResultSet.FETCH_FORWARD);
+            assertEquals(ResultSet.FETCH_FORWARD, rs.getFetchDirection());
+
+            assertEquals(ResultSet.CONCUR_READ_ONLY, rs.getConcurrency());
+            assertEquals(ResultSet.TYPE_FORWARD_ONLY, rs.getType());
+            assertEquals(ResultSet.HOLD_CURSORS_OVER_COMMIT, rs.getHoldability());
+
+            expectSqlException(rs::moveToCurrentRow, "The result set concurrency is CONCUR_READ_ONLY");
+        }
+    }
+
+    @ParameterizedTest
+    @EnumSource(names = {"PERIOD", "DURATION"}, mode = EnumSource.Mode.EXCLUDE)
+    public void wasNullPositional(ColumnType columnType) throws SQLException {
```

Comment:
let's also add test to record behavior of `wasNull` method which is called prior to any column access